### PR TITLE
Upgrade React version in all clients

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2076](https://ethyca.atlassian.net/browse/PROD-2076)

### Description Of Changes

Upgrade to React 18 along will all dependencies which need upgrading as a result.

### Code Changes

* lots of package upgrades
* lots of code tweaks as a result (most notably, the removal/migration of `React.FC` patterns and using Chakra v2s `isDisabled` for buttons)

### Steps to Confirm

* This will require a full regression
* Cypress tests passing
* Unit tests passing
* Smoke tests passing

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

[PROD-2076]: https://ethyca.atlassian.net/browse/PROD-2076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/admin-ui/src/features/privacy-notices/PrivacyNoticeTranslationForm.tsx [hunk 1]
```diff
@@ -190,7 +190,6 @@ const PrivacyNoticeTranslationForm = ({
         as={HStack}
         spacing={8}
         align="start"
-        grow={1}
```
Comment (gilluminate) [inline clients/admin-ui/src/features/privacy-notices/PrivacyNoticeTranslationForm.tsx:193]: this wasn't getting caught before, but is now. there is no `grow` prop

File: clients/admin-ui/src/features/datastore-connections/system_portal_config/types.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ export type AddConnectionStep = {
 };
 
 export type ConnectionConfigFormValues = {
-  description: string;
-  name: string;
+  description?: string;
+  name?: string;
```
Comment (gilluminate) [inline clients/admin-ui/src/features/datastore-connections/system_portal_config/types.ts:13]: these actually aren't always being set, which is causing errors

File: clients/admin-ui/src/features/privacy-requests/manual-processing/ManualAccessProcessingDetail.tsx [hunk 3]
```diff
@@ -24,18 +24,14 @@ import { PatchUploadManualWebhookDataRequest } from "privacy-requests/types";
...
 
-import { ManualInputData } from "./types";
+import { ManualProcessingDetailProps } from "./types";
 
-type ManualAccessProcessingDetailProps = {
-  connectorName: string;
-  data: ManualInputData;
-  isSubmitting: boolean;
-  onSaveClick: (params: PatchUploadManualWebhookDataRequest) => void;
-};
```
Comment (gilluminate) [inline clients/admin-ui/src/features/privacy-requests/manual-processing/ManualAccessProcessingDetail.tsx:34]: Exporting this from `./types` now to be able to share it with `ManualProcessingList.tsx`

File: clients/admin-ui/src/features/user-management/UserManagementTable.tsx [hunk 4]
```diff
@@ -13,10 +12,6 @@ import {
 } from "./user-management.slice";
 import UserManagementRow from "./UserManagementRow";
 
-interface UsersTableProps {
-  users?: User[];
```
Comment (gilluminate) [inline clients/admin-ui/src/features/user-management/UserManagementTable.tsx:17]: was no longer used

File: clients/admin-ui/src/features/common/CheckboxTree.tsx [hunk 5]
```diff
@@ -70,7 +70,7 @@ interface CheckboxItemProps {
   onExpanded: (node: TreeNode) => void;
   isIndeterminate: boolean;
   isDisabled: boolean;
-  children?: ReactNode;
+  children?: ReactNode[] | JSX.Element[] | null;
```
Comment (jpople) [inline clients/admin-ui/src/features/common/CheckboxTree.tsx:73]: Why was this necessary, out of curiosity?

**Review by jpople — APPROVED** [review]
Everything is passing for me!  Great work.

File: clients/admin-ui/src/features/common/CheckboxTree.tsx (hunk 5 above)
Comment (gilluminate) [reply clients/admin-ui/src/features/common/CheckboxTree.tsx]: great question. It was the fact that the `children` being passed was a ternary that sometimes resulted in `null`. I just took a new look at it and realized that if we return `undefined` instead, we don't need this extra effort on the typing. Good call out!
